### ISSUE TITLE
The shape of torch.mul is wrong?

### ISSUE THREAD

**Issue by luoyangbiao** [sbhonde1/Multimodal-Learning#6]
as the paper described, correlate dXC * CXd trans to dXd, but in your train.py, there is CXC ? if I got wrong operate output ? Can you tell me that, Thank you.

**Comment by wj-son** [comment]
def focal_reg_param(opt, loss_from, loss_to):

    diff = loss_from - loss_to
    return torch.exp(opt.beta*diff-1) if diff > 0 else torch.tensor([.0])


class ssa_loss(torch.nn.Module):
    def __init__(self):
        super(ssa_loss, self).__init__()

    def forward(self, m, n, rho=1):
        m, n = torch.mean(m, dim=0), torch.mean(n, dim=0)
        m, n = m.view(-1, m.shape[0]), n.view(-1, n.shape[0])
        m_corr, n_corr = torch.matmul(m, m.t()), torch.matmul(n, n.t())
        total_loss = rho * torch.pow(torch.norm(m_corr - n_corr), 2)
        return total_loss

I think this code is right.

**Comment by wj-son** [comment]
> as the paper described, correlate dXC * CXd trans to dXd, but in your train.py, there is CXC ? if I got wrong operate output ? Can you tell me that, Thank you.

I think this code is right. 

def focal_reg_param(opt, loss_from, loss_to):

    diff = loss_from - loss_to
    return torch.exp(opt.beta*diff-1) if diff > 0 else torch.tensor([.0])


class ssa_loss(torch.nn.Module):
    def __init__(self):
        super(ssa_loss, self).__init__()

    def forward(self, m, n, rho=1):
        m, n = torch.mean(m, dim=0), torch.mean(n, dim=0)
        m, n = m.view(-1, m.shape[0]), n.view(-1, n.shape[0])
        m_corr, n_corr = torch.matmul(m, m.t()), torch.matmul(n, n.t())
        total_loss = rho * torch.pow(torch.norm(m_corr - n_corr), 2)
        return total_loss

**Comment by wj-son** [comment]
I agree your suggestion. So, I made the code for the purpose.
~~~
def focal_reg_param(opt, loss_from, loss_to):

    diff = loss_from - loss_to
    return torch.exp(opt.beta*diff-1) if diff > 0 else torch.tensor([.0])

class ssa_loss(torch.nn.Module):
    def __init__(self):
        super(ssa_loss, self).__init__()

    def forward(self, m, n, rho=1):
        m, n = torch.mean(m, dim=0), torch.mean(n, dim=0)
        m, n = m.view(-1, m.shape[0]), n.view(-1, n.shape[0])
        m_corr, n_corr = torch.matmul(m, m.t()), torch.matmul(n, n.t())
        total_loss = rho * torch.pow(torch.norm(m_corr - n_corr), 2)
        return total_loss
~~~

**Comment by luoyangbiao** [comment]
could you share your source code with me of the paper: Improving the Performance of Unimodal Dynamic Hand-Gesture Recognition with Multimodal Training

**Comment by wj-son** [comment]
I'm sorry for saying that I have no source code. I just made core part of the method in the paper in order to apply that to my research (Human action recognition).

**Comment by sbhonde1** [comment]
Thank you @wj-son for pointing this out. 
I will make sure to update the code soon. I would love to accept your pull request if you have already solved this.

**Comment by wj-son** [comment]
Same as yours except for the two functions that I have attached. @sbhonde1 

But, I have some questions. In the paper, it is not clear that two or more optimizers and losses are used.
I have two suggestion. One is implemented in your code. Another is that parameters of both net_rgb and net_flow are put in a single optimizer. Next, total loss consists of all losses(loss_xross_rgb, loss_ssa_rgb, loss_xross_flow, loss_ssa_flow) and then the loss do backward().
I wonder why do you implement without the detailed implementation in the paper.
